tools/ossf_best_practices/silver_criteria.md

```diff
@@ -0,0 +1,338 @@
+
+_Possible answers: Met/Unmet/NA_
+
+
+## New functionality testing
+
+> The project MUST have a formal written policy that as major new functionality is added, tests for the new functionality MUST be added to an automated test suite.
+
+**Met**
+
+> The project MUST include, in its documented instructions for change proposals, the policy that tests are to be added for major new functionality.
+
+**Met**
+
+## Warning flags
+
+> Projects MUST be maximally strict with warnings in the software produced by the project, where practical.
+
+_Possible answers: Met/Unmet/NA_
+
+
+# Security
+
+## Secure development knowledge
+
+> The project MUST implement secure design principles (from "know_secure_design"), where applicable. If the project is not producing software, select "not applicable" (N/A).
+
+** Met**
+
+
+## Use basic good cryptographic practices
+
+_Note that some software does not need to use cryptographic mechanisms. If your project produces software that (1) includes, activates, or enables encryption functionality, and (2) might be released from the United States (US) to outside the US or to a non-US-citizen, you may be legally required to take a few extra steps. Typically this just involves sending an email. For more information, see the encryption section of [Understanding Open Source Technology & US Export Controls](https://www.linuxfoundation.org/resources/publications/understanding-us-export-controls-with-os-projects/)._
+
+> The default security mechanisms within the software produced by the project MUST NOT depend on cryptographic algorithms or modes with known serious weaknesses (e.g., the SHA-1 cryptographic hash algorithm or the CBC mode in SSH).
+
+_Possible answers: Met/Unmet/NA_
+
+> The project SHOULD support multiple cryptographic algorithms, so users can quickly switch if one is broken. Common symmetric key algorithms include AES, Twofish, and Serpent. Common cryptographic hash algorithm alternatives include SHA-2 (including SHA-224, SHA-256, SHA-384 AND SHA-512) and SHA-3.
+
+**Met**
+
+> The project MUST support storing authentication credentials (such as passwords and dynamic tokens) and private cryptographic keys in files that are separate from other information (such as configuration files, databases, and logs), and permit users to update and replace them without code recompilation. If the project never processes authentication credentials and private cryptographic keys, select "not applicable" (N/A).
+
+**N/A**
+
+> The software produced by the project SHOULD support secure protocols for all of its network communications, such as SSHv2 or later, TLS1.2 or later (HTTPS), IPsec, SFTP, and SNMPv3. Insecure protocols such as FTP, HTTP, telnet, SSLv3 or earlier, and SSHv1 SHOULD be disabled by default, and only enabled if the user specifically configures it. If the software produced by the project does not support network communications, select "not applicable" (N/A). 
+
+_Possible answers: Met/Unmet/NA_
```

Comment:
```suggestion
**Met**
```

Reply:
I think this could be N/A as by default there is no communicate endpoint enabled.

Reply:
Solved in 93bbe44